algebra/Atiyah.lean

```diff
@@ -0,0 +1,1695 @@
+import algebra.ring algebra.field data.set.basic order.zorn tactic.norm_num tactic.ring
+
+universes u v w
+
+-- page 1
+
+class zero_ring (α : Type u) [comm_ring α] : Prop :=
+(eq_zero : ∀ x:α, x = 0)
+
```

Comment:
I wonder if we need a separate class for this. We have `subsingleton` (which is also a `Prop`).
You could proof `[comm_ring α] → subsingleton α = (∀ x:α, x = 0)` and `[comm_ring α] → subsingleton α = (0 = 1)`.